Detections/CommonSecurityLog/Zscaler-LowVolumeDomainRequests

```diff
@@ -0,0 +1,49 @@
+id: 4d500e6d-c984-43a3-9f39-7edec8dcc04d
+name: Request for single resource on domain
+description: |
+  'This will look for connections to a domain where only a single file is requested, this is unusual as most modern web applications require additional recources. This type of activity is often assocaited with malware beaconing or tracking URL's delivered in emails. Developed for Zscaler but applicable to any outbound web logging.'
+severity: Low
+requiredDataConnectors:
+  - connectorId: Zscaler
+    dataTypes:
+      - CommonSecurityLog
+tactics:
+  - Command And Control
+relevantTechniques:
+  - T1102
+  - T1071
+query: |
+
+let timeRange = 7d;
+//The number of URI's seen to be suspicious, higher = less likely to be suspicious
+let uriThreshold = 1;
+let suspiciousURLs =
+CommonSecurityLog
+| where DeviceVendor == "Zscaler"
+// Only look at connections that were allowed through the web proxy. Comment line out for all events including those which were blocked
+| where DeviceAction == "Allowed"
+// Only look where some data was exchanged. Comment out to see 0 byte connections
+| where SentBytes > 0 and ReceivedBytes > 0
+//Extract the hostname
+| extend Domain = extract("([^.]*).{0,7}$", 1, DestinationHostName)
+| summarize makelist(RequestURL), makelist(TimeGenerated) by Domain
```

Comment:
Recommend that you use count() here to get your number of connections.  Calculating based on arraylength of TimeGenerated is more costly.  Recommended syntax - | summarize StartTimeUtc = min(TimeGenerated), EndTimeUtc = max(TimeGenerated), makelist(RequestURL), numOfConnections = count() by Domain